clickable/commands/build-libs.py

```diff
@@ -0,0 +1,62 @@
+import os
+import sys
+import subprocess
+import inspect
+import glob
+from os.path import dirname, basename, isfile, join
+
+from .base import Command
+from clickable.utils import print_warning
+from clickable.build_templates.base import Builder
+from clickable.container import Container
+from clickable.device import Device
+
+class LibBuildCommand(Command):
+    aliases = []
+    name = 'build-libs'
+    help = 'Compile the library dependencies'
+
+    def run(self, path_arg=None):
+        if not self.config.lib_configs:
+            print_warning('No libraries defined.')
+
+        self.container.setup_dependencies()
+
+        for lib in self.config.lib_configs:
+            dir_tmp = lib.dir
+            for arch in lib.architectures:
+                if arch in lib.arch_triplets:
+                    if not lib.custom_docker_image:
+                        lib.docker_image = 'clickable/ubuntu-sdk:16.04-{}'.format(arch)
+                    lib.dir = os.path.join(dir_tmp, lib.arch_triplets[arch])
+                    lib.arch = arch
+                    
+                    try:
+                        os.makedirs(lib.dir)
+                    except FileExistsError:
+                        pass
+                    except Exception:
+                        print_warning('Failed to create the build directory: {}'.format(str(sys.exc_info()[0])))
+
+                    if lib.prebuild:
+                        subprocess.check_call(lib.prebuild, cwd=self.config.cwd, shell=True)
+
+                    self.build(lib, Container(lib))
+
+                    if lib.postbuild:
+                        subprocess.check_call(lib.postbuild, cwd=lib.dir, shell=True)
+
+    def build(self, lib, container):
+        builder_classes = {}
+        builder_dir = join(dirname(__file__), '..')
+        modules = glob.glob(join(builder_dir, 'build_templates/*.py'))
+        builder_modules = [basename(f)[:-3] for f in modules if isfile(f) and not f.endswith('__init__.py')]
```

Comment:
It probably makes sense to abstract the loading of the builder classes into a function in utils since this is the second place it gets used.

Reply:
Done